flink-libraries/flink-python/src/main/python/org/apache/flink/python/api/flink/plan/Environment.py

```diff
@@ -163,10 +197,14 @@ def execute(self, local=False):
 
         The environment will execute all parts of the program that have resulted in a "sink" operation.
         """
+        global _operating, _last_env_id
         self._local_mode = local
         self._optimize_plan()
 
-        plan_mode = sys.stdin.readline().rstrip('\n') == "plan"
+        if self.container.is_executing():
+            plan_mode = False
+        else:
+            plan_mode = sys.stdin.readline().rstrip('\n') == "plan"
```

Comment:
we can move this logic into the container as well and get rid of the `plan_mode` variable.